### ISSUE TITLE
Issue with TFRecord from URL

### ISSUE THREAD

**Issue by mchantry** [ecmwf/climetlab#20]
I'm building a plugin for CML using TFRecord files.
When I point the MultiURL method at https://storage.ecmwf.europeanweather.cloud/MAELSTROM_AP3/TFR/TripCloud0.0.tfrecord
the file is successfully downloaded and loaded.
When I point it at https://storage.ecmwf.europeanweather.cloud/MAELSTROM_AP3/TFR/TripCloud0.1.tfrecord
the call to cml.load_source(....) fails with
```
Unknown file type /hugetmp/tmp-climetlab/url-63428528b56eb9c50418511287b38d1c7c53e6e203671861fb71faf86e6cbb72.man (b'PP\x00\x00\x00\x00\x00\x00'), ignoring
```
Examining the cache directory, the first file has been downloaded as a `.tfrecord` file and the second has been downloaded as a `.man` file.
Thanks.
Mat

**Comment by mchantry** [comment]
@floriankrb this still happens with 0.8.5.

**Comment by floriankrb** [comment]
Thanks for the test data.

This works for climetlab 0.8.6:

```python

s = cml.load_source('url', 'https://storage.ecmwf.europeanweather.cloud/MAELSTROM_AP3/TFR/EWCTest0.0.tfrecord')
s.to_tfdataset()

s = cml.load_source('url', 'https://storage.ecmwf.europeanweather.cloud/MAELSTROM_AP3/TFR/EWCTest0.1.tfrecord')
s.to_tfdataset()

```

**Comment by floriankrb** [comment]
I expect 0.8.6 version to be release  on pip in 30 min

**Comment by floriankrb** [comment]
But it still not work with you initial example.

The reason may be that the http header is providing the wrong data type (x-troff-man).

As you see (with `pip install httpie`):

```
$ http https://storage.ecmwf.europeanweather.cloud/MAELSTROM_AP3/TFR/TripCloud0.1.tfrecord
...
Content-Type: application/x-troff-man
```

```
$ http https://storage.ecmwf.europeanweather.cloud/MAELSTROM_AP3/TFR/EWCTest0.1.tfrecord
...
Content-Type: application/octet-stream
```

We will work on this shortly.

**Comment by floriankrb** [comment]
This issue definitely comes from the http server, but we can try some workarounds.

we could add an argument "extension" to load_source for urls. This would work as:

``` python
t = cml.load_source(
    "url",
    "https://storage.ecmwf.europeanweather.cloud/MAELSTROM_AP3/TFR/TripCloud0.1.tfrecord",
    extension="tfrecord",
)
t.to_tfdataset()
```

This would be experimental and may not work in future release (we will keep the functionaly if it is needed)

**Comment by floriankrb** [comment]
and this would be implemented with commit df7f5a5, current develop branch.

**Comment by floriankrb** [comment]
Works out-of-the box with version 0.8.9. No need for the "extension" argument.
